File: CS/lab1.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CS
{
    public class lab1
    {
        static void Main(string[] args)
        {
            List<Toy> toys = ReadFile("toys.txt");

            foreach (var toy in toys.Where(t => t.AgeRange == "4-10 years").OrderBy(t => t.Name))
            {
                Console.WriteLine($"{toy.Name} - {toy.Price} - {toy.AgeRange} - {toy.SpecialAttribute}");
            }

            Console.ReadKey();

        }
        public static List<Toy> ReadFile(string fileName)
        {
            var toys = new List<Toy>();

            try
            {
                using (StreamReader sr = new StreamReader(fileName))
                {
                    string line;
                    while ((line = sr.ReadLine()) != null)
                    {
                        string[] parts = line.Split(new string[] { ", " }, StringSplitOptions.None);
                        toys.Add(new Toy(parts[0], int.Parse(parts[1]), parts[2], parts[3]));
                    }
                }
            }
            catch (IOException e)
            {
                Console.WriteLine(e.Message);
            }

            return toys;
        }
    }
    public class Toy
    {
        public string Name { get; set; }
        public int Price { get; set; }
        public string AgeRange { get; set; }
        public string SpecialAttribute { get; set; }

        public Toy(string name, int price, string ageRange, string specialAttribute)
        {
            Name = name;
            Price = price;
            AgeRange = ageRange;
            SpecialAttribute = specialAttribute;
        }
    }
}

File: Java/src/lab1.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;
public class lab1 {
    public static void main(String[] args) {
        List<Toy> toys = readFile("toys.txt");

        toys.stream()
                .filter(toy -> toy.ageRange.equals("4-10 years"))
                .sorted(Comparator.comparing(toy -> toy.name))
                .forEach(toy -> {
                    System.out.println(toy.name + " - " + toy.price + " - " + toy.ageRange + " - " + toy.specialAttribute);
                });
    }
    public static List<Toy> readFile(String fileName) {
        List<Toy> toys = new ArrayList<>();

        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
            String line;
            while ((line = br.readLine()) != null) {
                String[] parts = line.split(", ");
                toys.add(new Toy(parts[0], Integer.parseInt(parts[1]), parts[2], parts[3]));
            }
        } catch (IOException e) {
            e.printStackTrace();
        }

        return toys;
    }
}

class Toy {
    String name;
    int price;
    String ageRange;
    String specialAttribute;

    public Toy(String name, int price, String ageRange, String specialAttribute) {
        this.name = name;
        this.price = price;
        this.ageRange = ageRange;
        this.specialAttribute = specialAttribute;
    }
}